tests/common/plugins/custom_skipif/__init__.py

```diff
@@ -0,0 +1,217 @@
+            skip_dictionary = yaml.load(skip_data, Loader=yaml.FullLoader)
+            item.session.config.cache.set(custom_skip_if_dict, skip_dictionary)
+    return skip_dictionary
+
+
+def get_tests_to_be_skipped_path(skip_tests_file='tests_to_be_skipped_conditionally.yaml'):
+    """
+    Get path to file with dynamic skip information
+    :param skip_tests_file: skip test file name
+    :return: full path to skip test file name
+    """
+    custom_skip_folder_path = os.path.dirname(__file__)
+    custom_skip_tests_file_path = os.path.join(custom_skip_folder_path, skip_tests_file)
+    return custom_skip_tests_file_path
+
+
+def update_syspath_for_dynamic_import():
+    """
+    Update sys.path by current folder to have possibility to load python modules dynamically
+    """
+    if os.path.dirname(__file__) not in sys.path:
+        sys.path.append(os.path.dirname(__file__))
+
+
+def prepare_checkers(skip_dict, current_platform):
+    """
+    Import dynamically checker modules and initialize them
+    :param skip_dict: dictionary with skip test case skip conditions
+    :param current_platform: string with current platform, example: x86_64-mlnx_msn2700-r0
+    :return: list with checkers objects
+    """
+    skip_checkers_list = []
+    extra_params = {'current_platform': current_platform}
+    for skip_by in skip_dict:
+        logger.debug('Importing dynamic skip module: {}'.format(skip_by))
+        try:
+            skip_module_obj = eval(
+                '__import__(skip_by).{}({}, {})'.format(skip_by, skip_dict[skip_by], extra_params))
+            skip_checkers_list.append(skip_module_obj)
+        except Exception as err:
+            logger.error('Unable to load dynamically skip object: {}'.format(err))
+    return skip_checkers_list
+
+
+def run_checkers_in_parallel(skip_checkers_list):
+    """
+    Run checkers in parallel and return results
+    :param skip_checkers_list: list with checkers objects
+    :return: dictionary with checkers result
+    """
+    manager = multiprocessing.Manager()
+    skip_dict_result = manager.dict()
+
+    proc_list = list()
+
+    for skip_check in skip_checkers_list:
+        skip_dict_result[skip_check.name] = None
+        proc_list.append(multiprocessing.Process(target=skip_check.is_skip_required, args=(skip_dict_result,)))
+
+    for proc in proc_list:
+        proc.start()
+    for proc in proc_list:
+        proc.join(timeout=60)
+
+    return skip_dict_result
+
+
+def make_skip_decision(skip_dict_result, current_skip_dict):
+    """
+    Make a final decision about whether to skip the test by combining the results of all the skip statements.
+    :param skip_dict_result: dictionary with checkers result
+    :param current_skip_dict: dictionary with skip test case data
+    :return: None or pytest.skip in case when need to skip test case
+    """
+    skip_reason = ''
+
+    skip_by_issue, skip_reason = is_skip_by_issue_required(skip_dict_result, skip_reason)
```

Comment:
I think `is_skip_by_issue_required` and `is_skip_by_platform_required` could be refactored to be a single generic function. This would greatly improve the extensibility of the system (each implementer should not have to modify this file).

If there is unique logic between the two I would suggest adding an abstract method to the `CustomSkipIf` class which does that and can be included in the subclasses.

Reply:
make sense, will do